source/include/FreeRTOS_Routing.h

```diff
@@ -0,0 +1,118 @@
+                                                                NetworkBufferDescriptor_t * const pxNetworkBuffer,
+                                                                BaseType_t xReleaseAfterSend );
+
+/* Return true as long as the LinkStatus on the PHY is present. */
+    typedef BaseType_t ( * GetPhyLinkStatusFunction_t ) ( struct xNetworkInterface * pxDescriptor );
+
+/** @brief These NetworkInterface access functions are collected in a struct: */
+    typedef struct xNetworkInterface
+    {
+        const char * pcName;                               /**< Just for logging, debugging. */
+        void * pvArgument;                                 /**< Will be passed to the access functions. */
+        NetworkInterfaceInitialiseFunction_t pfInitialise; /**< This function will be called upon initialisation and repeated until it returns pdPASS. */
+        NetworkInterfaceOutputFunction_t pfOutput;         /**< This function is supposed to send out a packet. */
+        GetPhyLinkStatusFunction_t pfGetPhyLinkStatus;     /**< This function will return pdTRUE as long as the PHY Link Status is high. */
+        struct
+        {
+            uint32_t
+                bInterfaceUp : 1,          /**< Non-zero as soon as the interface is up. */
+                bCallDownEvent : 1;        /**< The down-event must be called. */
+        } bits;                            /**< A collection of boolean flags. */
+
+        MACAddress_t xMACAddress;          /**< The MAC-address assigned to this end-point. */
+
+        struct xNetworkInterface * pxNext; /**< The next interface in a linked list. */
+    } NetworkInterface_t;
+
+/*
+ *  // As an example:
+ *  NetworkInterface_t xZynqDescriptor = {
+ *      .pcName					= "Zynq-GEM",
+ *      .pvArgument				= ( void * )1,
+ *      .pfInitialise           = xZynqGEMInitialise,
+ *      .pfOutput               = xZynqGEMOutput,
+ *      .pfGetPhyLinkStatus     = xZynqGEMGetPhyLinkStatus,
+ *  };
+ */
+
+
+/*
+ * Add a new physical Network Interface.  The object pointed to by 'pxInterface'
+ * must continue to exist.
+ * Only the Network Interface function xx_FillInterfaceDescriptor() shall call this function.
+ */
+    NetworkInterface_t * FreeRTOS_AddNetworkInterface( NetworkInterface_t * pxInterface );
+
+/*
+ * Get the first Network Interface.
+ */
+    NetworkInterface_t * FreeRTOS_FirstNetworkInterface( void );
+
+/*
+ * Get the next Network Interface.
+ */
+    NetworkInterface_t * FreeRTOS_NextNetworkInterface( const NetworkInterface_t * pxInterface );
+
+/* A ethernet packet has come in on a certain network interface.
+ * Find the best matching end-point. */
+    void FreeRTOS_MatchingEndpoint( const NetworkInterface_t * pxNetworkInterface,
+                                    NetworkBufferDescriptor_t * pucEthernetBuffer );
+
+
+    #include "FreeRTOS_Routing_IPv4.h"
+
+    #if ( ipconfigUSE_IPv6 != 0 )
```

Comment:
ipconfigUSE_IPv6 this has been updated to ipconfigUSE_IPV6 in FreeRTOSIPConfigDefaults.h file.